### PR TITLE
converting observable maps

### PR DESCRIPTION
**Pull request by quarrant**

### CODE REVIEW

**Comment by quarrant** [comment]
@codebelt review please

File: src/PersistStore.ts [hunk 1]
```diff
@@ -142,6 +153,10 @@ export class PersistStore<T, P extends keyof T> {
           actionPersistWarningIf(isActionProperty, propertyName);
 
           if (!isComputedProperty && !isActionProperty) {
+            if (target[propertyName] instanceof ObservableMap) {
+              target[propertyName] = Array.from(target[propertyName].entries());
```
Comment (codeBelt) [inline src/PersistStore.ts:157]: I don't think we want to write over the property in the store.

File: src/PersistStore.ts [hunk 2]
```diff
@@ -142,6 +153,10 @@ export class PersistStore<T, P extends keyof T> {
...
+            }
+
             propertiesToWatch[propertyName] = toJS(target[propertyName]);
```
Comment (codeBelt) [inline src/PersistStore.ts:160]: I think we want this?

```suggestion
            let propertyData = target[propertyName];
            
            if (propertyData instanceof ObservableMap) {
              propertyData = Array.from(propertyData.entries());
            }

            propertiesToWatch[propertyName] = toJS(propertyData);
```

File: src/PersistStore.ts [hunk 3]
```diff
@@ -105,6 +107,15 @@ export class PersistStore<T, P extends keyof T> {
...
+              if (target[propertyName] instanceof ObservableMap) {
+                const content = data[propertyName];
+
+                if (isArrayForMap(content)) {
+                  target[propertyName] = new Map(content);
+                  return;
+                }
+              }
+
               target[propertyName] = data[propertyName];
```
Comment (codeBelt) [inline src/PersistStore.ts:119]:
```suggestion
              const propertyData = data[propertyName];
              
              if (propertyData instanceof ObservableMap && isArrayForMap(propertyData)) {
                target[propertyName] = new Map(propertyData);
              } else {
                target[propertyName] = propertyData;
              }
```

Comment (quarrant) [reply src/PersistStore.ts:119]: You mistake here.

`target[propertyName] instanceof ObservableMap` is not equal `properyData instanceof ObservableMap`

File: src/PersistStore.ts (hunk 2 above)
Comment (quarrant) [reply src/PersistStore.ts:160]: it's OK

File: src/PersistStore.ts (hunk 3 above)
Comment (codeBelt) [reply src/PersistStore.ts:119]: your right, I updated the example above

**Pull request merged**